### PR TITLE
Support refund endpoints

### PR DESCRIPTION
**Pull request by ekoval**
Closes: #42

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: sitewit/services.py [hunk 1]
```diff
@@ -378,7 +378,7 @@ def cancel_search_campaign_subscription(self, account_token, campaign_id,
 
         Args:
             account_token (str): account token.
-            campaign_id (str): campaign to subscribe.
+            campaign_id (str): campaign to refund.
```
Comment (RayeN) [inline sitewit/services.py]: Maybe `to cancel` ?

File: tests/campaigns/test_integration.py [hunk 2]
```diff
@@ -262,6 +266,42 @@ def test_error_404_is_raised(self):
...
+    def setUpClass(cls):
+        super(TestCancelPrePurchasedSubscriptionTestCase, cls).setUpClass()
+
+    @classmethod
+    def create_campaign(cls):
+        # Pre-purchased campaign created "on the fly", when we subscribe
+        # to it.
+        return cls.subscribe_method(cls.account_token, -1, 500, 'USD')['id']
+
+    def test_refund_request_is_accepted(self):
```
Comment (RayeN) [inline tests/campaigns/test_integration.py]: wrong test name?

**Comment by RayeN** [comment]
Can you please add a test for canceling pre-purchased campaign?

**Comment by ekoval** [comment]
> Can you please add a test for canceling pre-purchased campaign?

It is added: `class TestCancelPrePurchasedSubscriptionTestCase(`

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: tests/campaigns/test_integration.py [hunk 3]
```diff
@@ -262,6 +266,42 @@ def test_error_404_is_raised(self):
...
+        self.assertEqual(self.result['campaignInfo']['status'], 'Cancelled')
+        self.assertEqual(self.result['refundAmount'], -500.0)
+
+
+class TestCancelPrePurchasedSubscriptionTestCase(
+        BaseCampaignTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestCancelPrePurchasedSubscriptionTestCase, cls).setUpClass()
```
Comment (RayeN) [inline tests/campaigns/test_integration.py]: we don't do anything except for calling super, maybe just remove it?

File: sitewit/services.py [hunk 4]
```diff
@@ -418,6 +417,46 @@ def cancel_display_campaign_subscription(self, account_token, campaign_id,
...
+    def refund_display_campaign_subscription(self, account_token, campaign_id):
+        """Cancel display subscription and initiate refund.
+
+        Cancels a pre-purchased display campaign's subscription and issues
+        a refund for accrued spend. This will only work for campaigns that
+        have not launched.
+
+        Args:
+            account_token (str): account token.
+            campaign_id (str): campaign to refund.
```
Comment (RayeN) [inline sitewit/services.py]: `cancel/refund`? just to match other docstring

File: sitewit/services.py [hunk 5]
```diff
@@ -378,7 +377,7 @@ def cancel_search_campaign_subscription(self, account_token, campaign_id,
 
         Args:
             account_token (str): account token.
-            campaign_id (str): campaign to subscribe.
+            campaign_id (str): campaign to cancel/refund.
```
Comment (RayeN) [inline sitewit/services.py]: We don't do refund here, don't we?

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by RayeN** [comment]
I think we should investigate why tests are failing... But probably out of the scope of this pull.

**Comment by RayeN** [comment]
Can you please create a new ticket for that?

**Comment by ekoval** [comment]
> I think we should investigate why tests are failing... But probably out of the scope of this pull.

The problem is that I have no idea how to investigate this. Will file a ticket.

**Pull request merged**